packages/react/src/components/CodeSnippet/__tests__/CodeSnippet-test.js

```diff
@@ -6,102 +6,188 @@
+    "doctoc": "doctoc --title '## Table of Contents'",
+    "format": "prettier --write '**/*.{js,md,scss,ts}' '!**/{build,es,lib,storybook,ts,umd}/**'",
+    "format:diff": "prettier --list-different '**/*.{js,md,scss,ts}' '!**/{build,es,lib,storybook,ts,umd}/**' '!packages/components/**'",
+    "lint": "eslint actions config codemods packages",
+    "lint:styles": "stylelint '**/*.{css,scss}' --report-needless-disables --report-invalid-scope-disables",
+    "sync": "carbon-cli sync",
+    "test": "cross-env BABEL_ENV=test jest",
+    "test:e2e": "cross-env BABEL_ENV=test jest --testPathPattern=e2e --testPathIgnorePatterns='examples,/packages/components/,/packages/react/'"
+  },
+  "resolutions": {
+    "react": "~16.9.0",`;
+
+describe('CodeSnippet', () => {
+  it('should use the appropriate snippet class when it is type single', () => {
+    render(
+      <CodeSnippet type="single" data-testid="code-1">
+        {single}
+      </CodeSnippet>
+    );
+    expect(screen.getByTestId('code-1')).toHaveClass('cds--snippet--single');
+  });
+
+  it('should use the appropriate snippet class when it is type multi', () => {
+    render(
+      <CodeSnippet type="multi" data-testid="code-2">
+        {multiLong}
+      </CodeSnippet>
+    );
+
+    expect(screen.getByTestId('code-2')).toHaveClass('cds--snippet--multi');
+  });
+
+  it('should use the appropriate snippet class when it is type inline', () => {
+    render(
+      <CodeSnippet type="inline" data-testid="code-3">
+        {inline}
+      </CodeSnippet>
+    );
+
+    expect(screen.getByTestId('code-3')).toHaveClass('cds--snippet--inline');
   });
 
-  describe('Triggers appropriate events', () => {
-    it('should call the click handler', () => {
-      const onClick = jest.fn();
-      const clickWrapper = mount(<CodeSnippet onClick={onClick} />);
-      clickWrapper.find(CopyButton).simulate('click');
-      expect(onClick).toHaveBeenCalled();
-    });
-
-    it('should call the click handler with type="inline"', () => {
-      const onClick = jest.fn();
-      const clickWrapper = mount(
-        <CodeSnippet type={'inline'} onClick={onClick} />
-      );
-      clickWrapper.find(Copy).simulate('click');
-      expect(onClick).toHaveBeenCalled();
-    });
+  it('should render children as expected', () => {
+    render(
+      <CodeSnippet type="inline" data-testid="code-4">
+        {inline}
+      </CodeSnippet>
+    );
+
+    expect(screen.getByTestId('code-4')).toHaveTextContent(inline);
   });
 
-  describe('check for showMoreBtn', () => {
-    let wrapper;
+  it('should allow custom classes to be applied when passed in via className', () => {
+    render(
+      <CodeSnippet type="inline" data-testid="code-5" className="custom-class">
+        {inline}
+      </CodeSnippet>
+    );
 
-    beforeEach(() => {
-      wrapper = mount(<CodeSnippet type={'multi'} />);
-    });
+    expect(screen.getByTestId('code-5')).toHaveClass('custom-class');
```

Comment:
I think for `className` placement we should be explicit in heirarchy and use `container`:

```jsx
  it('should allow custom classes to be applied when passed in via className', () => {
    const { container } = render(
      <CodeSnippet type="inline" className="custom-class">
        {inline}
      </CodeSnippet>
    );

    expect(container.firstChild).toHaveClass('custom-class');
```

This can help us catch if `className` is moved to a different part of the component